### DISCUSSION TITLE
Google Cloud xDrip - NightscoutUploader fail (SSL Handshake Exception)

### DISCUSSION THREAD

**Discussion by buxxdev** [NightscoutFoundation/xDrip#2510] [Q&A]
Hi there,
I've deployed Nightscout in Google Cloud using the xdrip+ teams excellent installer but after 24 hours of stable performance this morning when my daughter (who's phone is running xdrip+) went to school her readings stopped reporting into Nightscout, although Dexcom Share which xdrip+ is also uploading into was fine.  It conincided with her moving onto 4G coverage as she left the house but this may be coincidental.
Then it started working again by itself around 2pm without any intervention.
The nightscout site was up and when she got home I've checked the xdrip logs which is showing some SSLHandshake Exception as shown in the screen capture below:

![image](https://user-images.githubusercontent.com/55636308/204370955-ca06fcb0-84f8-4aa8-b561-45658be9ca99.png)

I logged into the google cloud instance and checked the nginx and letsencrypt logs in /var/log/ and nothing is standing out as an issue.

Any thoughts?

It may be fine now but really strange it fixed itself without understanding why....

**Comment by Navid200** [comment]
Can you check on the following page?

Settings &#8722;> Cloud Upload &#8722;> Nightscout Sync

Is "Use mobile data" enabled on that page?

**Reply by buxxdev** [reply]
Yes it’s enabled

**Comment by Navid200** [comment]
What happened around 2?   Is that when your daughter left school?

**Reply by buxxdev** [reply]
so actually I checked the Nginx access.log on the Nightscout server and xdrip+ started uploading again after about 3 hours (10:52) - I wasn't quite sure when it started again as it backfilled.
So quite strange as the SSL handshake issue was still showing in the app logs.

![image](https://user-images.githubusercontent.com/55636308/204393507-29612b38-b40f-4b0e-9148-23e7b5d4a2fa.png)

2pm she was still at school and I can see POSTs from the same IP so nothing changed connectivity wise.
I'll monitor it again tomorrow as its been stable again since 2pm on mobile data and wifi at home.

**Reply by Navid200** [reply]
I am not sure what's going on.  So, I am very curious.  I was just thinking there may be places, like schools, that would block access to some particular web sites.  I don't know why.  But, may be the school blocks access to FreeDNS.  I hope not.

**Reply by buxxdev** [reply]
I’m curious too…. She doesn’t use the school WiFi so it can’t be that.  I wondered if her data plan has some content filter baring the freeDNS domain but then it randomly started working.  I thought the fact that the SSL handshake issue was happening meant e2e connectivity was in place but I’m not sure not as it started working while those errors were still there.  I’ll see how tomorrow goes and update here…

**Reply by buxxdev** [reply]
Does the freeDNS script have a log? I’m wondering if the Google cloud instance IP changed and there was a DNS cache somewhere

**Reply by buxxdev** [reply]
Its not a DNS issue, i checked the DNS history and the IP had not changed...

Same has happened this morning which is strange.  I should have tested moving the phone onto 4G last night to replicate it.
I installed tcpdump and took a pcap from the the google cloud server and seems seems there is communication from her phone and the server but the phone is sending a TCP RST so somethings not right at the lower level.  Seems its just looping around SYN -> SYN,ACK -> ACK and then finally RST,ACK.

I rebooted the server and issue is still there.

**Reply by tzachi-dar** [reply]
Your case is very weird, but someone else has complained that there is a problem with freedns.
You might want to try and move to another dns provider in order to see if the problem still exists.
See my comment on https://github.com/NightscoutFoundation/xDrip/discussions/2441#discussioncomment-4249763 on how to do it.

**Reply by buxxdev** [reply]
Thanks and I agree it’s very weird!!
I’ll do some more tracing when she’s home and I have the phone tonight.
I agree moving to another DNS might solve the issue

**Reply by buxxdev** [reply]
I believe the problems solved….
It seems the domain privatedns.org is black listed by my mobile network as and adult website and the subscription for my daughters phone had an adult content bar applied.
I removed the adult bar and now it seems it’s working fine 🤞(this is fine for me as my daughter doesn’t use the phone as she’s only 5 but may not be appropriate for others).

Why it worked yesterday after when she was still at school I have no idea.

So it seems there are no issues with xdrip+ working with the Google Cloud installation but others should be cautious that these dynastic DNS domains might be black listed as adult content sites by mobile operators.

Thanks for the support as always!
